src/main/java/de/tum/in/www1/artemis/repository/ExerciseRepository.java

```diff
@@ -69,6 +69,52 @@ public interface ExerciseRepository extends JpaRepository<Exercise, Long> {
             """)
     Set<Exercise> findAllExercisesWithCurrentOrUpcomingDueDate(@Param("now") ZonedDateTime now);
 
+    @Query("""
+            SELECT COUNT(e.id)
+            FROM Exercise e
+            WHERE e.course.testCourse = FALSE
+            	AND e.dueDate >= :#{#now}
+            	AND e.dueDate <= :#{#maxDate}
+                AND e.class = :#{#exerciseType}
+            """)
+    Integer countExercisesWithCurrentOrUpcomingDueDateWithinTimeRange(@Param("now") ZonedDateTime now, @Param("maxDate") ZonedDateTime maxDate,
+            @Param("exerciseType") String exerciseTypeDiscriminator);
```

Comment:
We also use complaintType in the ComplaintRepository. Why can't we use the normal Enum here?

Reply:
Maybe I missed something, but isn't the `ComplaintType` just a regular Enum whereas the `exerciseType` is part of the polymorphic relation definition?

Reply:
You're right, Complaints do not use subclasses. But the exercise subclasses still should have the ExerciseType set, don't they?
But if it's not possible here or you think this is currently a better solution, feel free to leave it like that. The question was mainly out of curiosity